File: src/main/java/gargoyle/calendar/util/resources/ClasspathResource.java
package gargoyle.calendar.util.resources;

import java.net.URL;

public class ClasspathResource extends ResourceBase {
    public ClasspathResource(String location) {
        this(location, ClasspathResource.class.getClassLoader());
    }

    public ClasspathResource(String location, ClassLoader classLoader) {
        super(Resources.getUrl(location, classLoader));
    }

    public ClasspathResource(String location, Class<?> aClass) {
        super(Resources.getUrl(location, aClass));
    }

    public ClasspathResource(URL location) {
        super(location);
    }


}

File: cal-core/src/main/kotlin/gargoyle/calendar/util/resources/ClasspathResource.kt
package gargoyle.calendar.util.resources

import java.net.URL
import kotlin.reflect.KClass

class ClasspathResource : ResourceBase {
    constructor(location: String, classLoader: ClassLoader)
            : super(Resources.getUrl(location, classLoader))

    constructor(location: String, type: KClass<*>) : super(Resources.getUrl(location, type))
    constructor(location: URL) : super(location)
}
